src/index.js

```diff
@@ -158,7 +158,7 @@ function createLoadableComponent(loadFn, options) {
       return init();
     }
 
-    componentWillMount() {
+    componentDidMount() {
       this._mounted = true;
       this._loadModule();
```

Comment:
should call `_loadModule` in `constructor`

Reply:
And handle it:
https://github.com/jamiebuilds/react-loadable/pull/195/files#diff-1fdf421c05c1140f6d71444ea2b27638R198-R202

Reply:
Do you recommend where to split the logic? obs cant call setState in the constructor, but _loadModule is also used in the retry method

Reply:
Slightly ugly, but you could make an internal `_ setState`

```js
_setState(state) {
  if (this._mounted) {
    this.setState(state);
  } else {
    this.state = state;
  }
}
```

Reply:
Actually I tried this and React will still warn about causing SideEffects.

So, I don't think `_loadModule` (as-is) should be called in the `constructor` as suggested.

If you want to try it, it's here and on NPM (but I wouldn't recommend it)
https://github.com/axiomhq/react-loadable

Reply:
Probably better to rename it to `UNSAFE_ componentWillMount` and call it a day

Reply:
I ended up trying out this change (using `componentDidMount`) in `@axiomhq/react-loadable@5.5.2`.

It also has side effect warnings.

It looks like the new React API is just too different to make this easily work.

![image](https://user-images.githubusercontent.com/404765/64618144-56c00480-d3a5-11e9-91f0-e612b6f4bd53.png)